src/dht.cpp

```diff
@@ -941,8 +946,20 @@ Dht::searchStep(Search& sr)
     DHT_DEBUG("[search %s IPv%c] step", sr.id.toString().c_str(), sr.af == AF_INET ? '4' : '6');
     sr.step_time = now;
 
+    /*
+     * The accurate delay between two refills has not been strongly determined.
+     * TODO: Emprical analysis over refill timeout.
+     */
+    if (sr.refill_time + Node::NODE_EXPIRE_TIME < now and sr.nodes.size()-sr.getNumberOfBadNodes(now) < SEARCH_NODES) {
```

Comment:
You can refer to #30 for brief and vague reminder. The following should be more complete:

*The goal was to fasten the "refill" of a search, i.e. the replacement of bad nodes in a wider range around the target hash of the search.*

Before this, the refill of a search was done upon the expiration of a `Search` (all nodes in a search would be expired). We decided to enhance the solution by using the approach where a `Dht::Search::refill` would be done each call of `Dht::Searchstep` always if *last refill* was made at least 10 minutes before. A delay was necessary in case of critical scenario where a considerable amount of nodes would expire all around the target hash of the Search, therefor over refilling the search with potentially wrong nodes (i.e. too far from target hash compared to others). This scenario could be observed in one of our benchmark tests (`benchmark.py --persistence -t delete`, or `-t replace`). This change would make the refill progressive and not wait for a `Search` to expire.

As said in the comment at line 950, the wait time has been arbitrary determined. However, I'm mostly certain there was a reasoning behind choosing 10 minutes, but I can't remember right now. But as you're saying